### PR TITLE
Fix demos on Marvell

### PR DESCRIPTION
**Pull request by leegeth**
Fix demos on Marvell

Description
-----------
Problem: Many demos were failing in Marvell board due to lack of memory in heap while establishing a TLS connection.
Root Cause: Establishing a TLS connection requires a good amount of memory in the HEAP. The tasks created with xTaskCreate() function also allocates the memory for the stack in the heap. Marvell's heap(~73KB) was running out of space due to these allocations.
Fix: Optimized the stack sizes for different tasks for the demo and verified the demos.

The demo verification CI job can be found here
https://amazon-freertos-ci.corp.amazon.com/job/im_v2_master/job/nightly_custom_job/1312/

The stack high water mark are analyzed using the jobs
https://amazon-freertos-ci.corp.amazon.com/job/im_v2_master/job/nightly_custom_job/1356/
https://amazon-freertos-ci.corp.amazon.com/job/im_v2_master/job/nightly_custom_job/1358/
https://amazon-freertos-ci.corp.amazon.com/job/im_v2_master/job/nightly_custom_job/1362/

Please find the minimum value of the stack high water mark for each of the tasks in different demo runs below.
Taskpool task - 727
Network receive task - 804
GGD Demo task - 531
Logging task - 409
Shadow task - 423

Checklist:
----------
- [x] I have tested my changes. No regression in existing tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/marvell/boards/mw300_rd/aws_demos/config_files/iot_config.h [hunk 1]
```diff
@@ -37,20 +37,21 @@
  * level for all libraries; the library-specific settings override the global
  * setting. If both the library-specific and global settings are undefined,
  * no logs will be printed. */
-#define IOT_LOG_LEVEL_GLOBAL                    IOT_LOG_INFO
```
Comment (abhidixi11) [inline vendors/marvell/boards/mw300_rd/aws_demos/config_files/iot_config.h:40]: Do we need these log level changes ?

Comment (leegeth) [reply vendors/marvell/boards/mw300_rd/aws_demos/config_files/iot_config.h:40]: Yes. Reducing the log levels also helped in reducing the memory need in Heap. The logging task mallocs buffers for handling logs.

**Review by abhidixi11 — DISMISSED** [review]

File: vendors/marvell/boards/mw300_rd/aws_demos/application_code/main.c [hunk 2]
```diff
@@ -57,7 +57,7 @@ const AppVersion32_t xAppFirmwareVersion = {
 
 /* Logging Task Defines. */
 #define mainLOGGING_MESSAGE_QUEUE_LENGTH    ( 32 )
-#define mainLOGGING_TASK_STACK_SIZE         ( configMINIMAL_STACK_SIZE * 8 )
+#define mainLOGGING_TASK_STACK_SIZE         ( configMINIMAL_STACK_SIZE * 6 )
```
Comment (n9wxu) [inline vendors/marvell/boards/mw300_rd/aws_demos/application_code/main.c:60]: Giving the logging task stack size in terms of multiples of minimal stack size is not helpful.  Just put in the number you want.  stack size is a set of educated guesses so lets not pretend this is the science to the method.

File: vendors/marvell/boards/mw300_rd/aws_demos/config_files/iot_config.h [hunk 3]
```diff
@@ -37,20 +37,21 @@
...
-#define AWS_IOT_LOG_LEVEL_DEFENDER              IOT_LOG_INFO
-#define IOT_LOG_LEVEL_HTTPS                     IOT_LOG_INFO
+#define IOT_LOG_LEVEL_MQTT                      IOT_LOG_ERROR
+#define AWS_IOT_LOG_LEVEL_SHADOW                IOT_LOG_ERROR
+#define AWS_IOT_LOG_LEVEL_DEFENDER              IOT_LOG_ERROR
+#define IOT_LOG_LEVEL_HTTPS                     IOT_LOG_ERROR
 
 /* Platform thread stack size and priority. */
-#define IOT_THREAD_DEFAULT_STACK_SIZE           3840
+#define IOT_THREAD_DEFAULT_STACK_SIZE           1024
```
Comment (n9wxu) [inline vendors/marvell/boards/mw300_rd/aws_demos/config_files/iot_config.h:51]: putting in numbers makes sense.  This is nice

**Review by n9wxu — CHANGES_REQUESTED** [review]
Please make the stack sizes actual numbers instead of some sort of multiple.

File: vendors/marvell/boards/mw300_rd/aws_demos/config_files/iot_config.h (hunk 3 above)
Comment (leegeth) [reply vendors/marvell/boards/mw300_rd/aws_demos/config_files/iot_config.h:51]: Thanks

File: vendors/marvell/boards/mw300_rd/aws_demos/application_code/main.c (hunk 2 above)
Comment (leegeth) [reply vendors/marvell/boards/mw300_rd/aws_demos/application_code/main.c]: Addressed it. Thanks.

**Review by n9wxu — APPROVED** [review]

**Review by abhidixi11 — APPROVED** [review]

**Review by abhidixi11 — APPROVED** [review]

**Pull request merged**